receiver/datadogreceiver/internal/translator/traces_translator.go

```diff
@@ -165,6 +168,56 @@ func ToTraces(payload *pb.TracerPayload, req *http.Request) ptrace.Traces {
 	return results
 }
 
+// DDSpanLink represents the structure of each JSON object
+type DDSpanLink struct {
+	TraceID    string         `json:"trace_id"`
+	SpanID     string         `json:"span_id"`
+	Tracestate string         `json:"tracestate"`
+	Attributes map[string]any `json:"attributes"`
+}
+
+func tagsToSpanLinks(tags map[string]string, dest ptrace.SpanLinkSlice) error {
+	key := "_dd.span_links"
+	val, ok := tags[key]
+	if !ok {
+		return nil
+	}
+	delete(tags, key)
+
+	var spans []DDSpanLink
+	err := json.Unmarshal([]byte(val), &spans)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(spans); i++ {
+		span := spans[i]
+		link := dest.AppendEmpty()
+
+		// Convert trace id.
```

Comment:
You can use something like this instead:
```go
		// Convert trace id.
		rawTrace, errTrace := trace.TraceIDFromHex(span.TraceID)
		if errTrace != nil {
			return errTrace
		}
		link.SetTraceID(pcommon.TraceID(rawTrace))
```

`trace` is `go.opentelemetry.io/otel/trace`